Library/Homebrew/cmd/update-bash.sh

```diff
@@ -198,45 +243,43 @@ update-bash() {
       --simulate-from-current-branch) HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH=1 ;;
       --*) ;;
       -*v*) HOMEBREW_VERBOSE=1 ;;
-      -*v*) HOMEBREW_DEBUG=1 ;;
+      -*d*) HOMEBREW_DEBUG=1 ;;
       -*) ;;
       *)
-        echo "This command updates brew itself, and does not take formula names." >&2
-        echo "Use 'brew upgrade <formula>'." >&2
-        exit 1
+        odie <<-EOS
+This command updates brew itself, and does not take formula names.
+Use 'brew upgrade <formula>'."
+EOS
         ;;
     esac
   done
 
-  if [ -n "$HOMEBREW_DEBUG" ]
+  if [[ -n "$HOMEBREW_DEBUG" ]]
   then
     set -x
   fi
 
   # check permissions
-  if [ "$HOMEBREW_PREFIX" = "/usr/local" ] && ! test -w /usr/local
+  if [[ "$HOMEBREW_PREFIX" = "/usr/local" && ! -w /usr/local ]]
   then
-    echo "Error: /usr/local must be writable!" >&2
-    exit 1
+    odie "/usr/local must be writable!"
   fi
 
-  if ! test -w "$HOMEBREW_REPOSITORY"
+  if [[ ! -w "$HOMEBREW_REPOSITORY" ]]
   then
-    echo "Error: $HOMEBREW_REPOSITORY must be writable!" >&2
-    exit 1
+    odie "$HOMEBREW_REPOSITORY must be writable!"
   fi
 
-  if [ -z "$(which_git)" ]
+  if [[ -z "$(which_git)" ]]
   then
     brew install git
-    if [ -z "$(which_git)" ]
+    if [[ -z "$(which_git)" ]]
     then
-      echo "Error: Git must be installed and in your PATH!" >&2
-      exit 1
+      odie "Git must be installed and in your PATH!"
     fi
   fi
 
-  if [ -z "$HOMEBREW_VERBOSE" ]
+  if [[ -z "$HOMEBREW_VERBOSE" ]]
   then
     QUIET_ARGS="-q"
```

Comment:
What do you think about converting `QUIET_ARGS` into an array? It would be initially `QUIET_ARGS=()` and overridden in this line to be `QUIET_ARGS=(-q)`. It would then be used via `"${QUIET_ARGS[@]}"`, the advantage being that `shellcheck` will no longer complain about it being used unquoted (and we can remove the comments that currently suppress the check).